src/data/preprocessing.py

```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+import numpy as np
+from sklearn.preprocessing import LabelEncoder, StandardScaler, OneHotEncoder
+from sklearn.pipeline import Pipeline
+from sklearn.impute import SimpleImputer
+from sklearn.compose import ColumnTransformer
+
+import src.data.utils as utils
+
+
+class DataPreprocessor:
+    ''' Class for reading dataset file from a given path, processing it
+        properly and saving to given path as a csv file.
+    '''
+    def __init__(self):
+        self.X, self.y = pd.DataFrame(), pd.DataFrame()
+        self.dataset_name = str()
+
+    def read_data(self, path: str):
+        ''' Reads raw data into DataProcessor '''
+
+        # Pull out the dataset name from the path
+        self.dataset_name = path.split('/')[-1].split('.')[0]
+        preprocessing_function = utils.get_preprocessing_function(self.dataset_name)
+        self.X, self.y = preprocessing_function(path)
+
+    def preprocess_data(self):
+        ''' Preprocess data using defined pipelines for categorical and
+            numerical features
+        '''
+        if self.dataset_name == 'drugs':
```

Comment:
This hard-coding of dataset's name for special treatment in the pipeline can become problematic in the future (it constitutes a technical debt), please make sure that this is absolutely required.